File: lib/domain/repositories/activity_log_repository.dart
import 'package:maki_mobile_pos/domain/entities/entities.dart';

/// Abstract repository contract for ActivityLog operations.
abstract class ActivityLogRepository {
  /// Logs an activity.
  Future<ActivityLogEntity> logActivity(ActivityLogEntity log);

  /// Gets activity logs. An empty [types] means "every operation".
  Future<List<ActivityLogEntity>> getActivityLogs({
    List<ActivityType> types = const [],
    DateTime? startDate,
    DateTime? endDate,
    int limit = 50,
  });
}

File: web_admin/src/domain/repositories/ActivityLogRepository.ts
// Mirror of lib/domain/repositories/activity_log_repository.dart's read
// surface. Reads are one-shot: /admin/logs fetches only when the admin
// submits filters, so there is no live-subscription method here.

import type { ActivityLog, ActivityType } from '../entities';

export interface ActivityLogQuery {
  /** Empty or omitted means "every operation". */
  types?: ActivityType[];
  start?: Date;
  end?: Date;
  limit?: number;
}

export interface ActivityLogRepository {
  list(query?: ActivityLogQuery): Promise<ActivityLog[]>;
  log(input: Omit<ActivityLog, 'id' | 'createdAt'>): Promise<void>;
}
